loadtest/store/memstore/random.go

```diff
@@ -448,3 +449,70 @@ func (s *MemStore) RandomDraftForTeam(teamId string) (string, error) {
 
 	return draftIDs[rand.Intn(len(draftIDs))], nil
 }
+
+func (s *MemStore) GetRandomScheduledPost() (*model.ScheduledPost, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	// Seed the random generator
+	rand.Seed(time.Now().UnixNano())
+
+	// Check if scheduledPosts is empty
+	if len(s.scheduledPosts) == 0 {
+		return nil, errors.New("no scheduled posts available")
+	}
+
+	// Pick a random index for the outer map
+	randomOuterIndex := rand.Intn(len(s.scheduledPosts))
+	var selectedInnerMap map[string]*model.ScheduledPost
+	outerIndex := 0
+	for _, innerMap := range s.scheduledPosts {
+		if outerIndex == randomOuterIndex {
+			selectedInnerMap = innerMap
+			break
+		}
+		outerIndex++
+	}
+
+	// Check if the selected inner map is empty
+	if len(selectedInnerMap) == 0 {
+		return nil, errors.New("no posts available in selected category")
+	}
```

Comment:
I think we shouldn't error out if there are no posts in a single inner map. In that case, we should pick another one. Maybe what we can do is: build a slice of all keys whose inner map contain at least one element. Then, get a random element from that slice, and then get a random post from that inner map. Something like the following (not tested):
```go
var keys []string
for key, innerMap := range s.scheduledPosts {
	if len(innerMap) > 0 {
		keys = append(keys, key)
	}
}

if len(keys) == 0 {
	return nil, errors.New("no posts available")
}

selectedInnerMap := s.scheduledPosts[keys[rand.Intn(len(keys))]

Reply:
Done.